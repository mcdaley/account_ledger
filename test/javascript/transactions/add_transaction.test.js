//-----------------------------------------------------------------------------
// test/javascripts/ledger/ledger.test.js
//-----------------------------------------------------------------------------
import React                from 'react'
import ReactDOM             from 'react-dom'
import renderer             from 'react-test-renderer'
import {shallow, configure} from 'enzyme'
import Adapter              from 'enzyme-adapter-react-16'
configure({ adapter: new Adapter() })

import AddTransaction       from '../../../app/javascript/components/transactions/add_transaction'

///////////////////////////////////////////////////////////////////////////////
// TODO: 02/26/2018
// [] Test layout of AddTransaction Form
//    1.) Snapshot of component
//    2.) Test calling Submit/OnClick
// [] Test clicking 'Add' with empty form, should see errors
//    1.) Simulate AJAX response using a sinon.stub()
//    2.) Verify the empty fields have class '.has-error'
// [] Test clicking 'Add' with fields, should receive new transaction 
///////////////////////////////////////////////////////////////////////////////


it('renders correctly', () => {
  const handleSubmitMock = jest.fn()
  
  const tree = renderer
    .create(<AddTransaction addTransaction={handleSubmitMock} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
});

/**
it('handles submit', () => {
  const handleSubmitMock  = jest.fn()
  const wrapper           = shallow(<AddTransaction addTransaction={handleSubmitMock} />)
  wrapper.find('.btn-submit').simulate('click', { preventDefault: () => {} })
  expect(handleSubmitMock.mock.calls.length).toBe(1)
})
**/


