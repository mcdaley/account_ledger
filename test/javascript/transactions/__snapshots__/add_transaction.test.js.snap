// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="card bg-info"
  style={
    Object {
      "marginBottom": "0.5rem",
    }
  }
>
  <form
    className="card-body"
    style={
      Object {
        "paddingBottom": "0rem",
        "paddingTop": "0.5rem",
      }
    }
  >
    <div
      className="form-row"
    >
      <div
        className="col-3"
      >
        <label
          className="sr-only"
        >
          Date
        </label>
        <input
          className="form-control"
          onChange={[Function]}
          type="date"
          value=""
        />
      </div>
      <div
        className="col-5"
      >
        <label
          className="sr-only"
        >
          Description
        </label>
        <input
          className="form-control"
          id="txnDescription"
          onChange={[Function]}
          placeholder="Description"
          type="text"
          value=""
        />
      </div>
      <div
        className="col-2"
      >
        <label
          className="sr-only"
        >
          Amount
        </label>
        <input
          className="form-control"
          id="txnAmount"
          onChange={[Function]}
          placeholder="Amount"
          type="text"
          value=""
        />
      </div>
      <div
        className="col-2"
      >
        <button
          className="btn btn-primary mb-2 btn-submit"
          onClick={[Function]}
          type="submit"
        >
           Add
        </button>
        <button
          className="btn mb-2"
          onClick={[Function]}
          style={
            Object {
              "marginLeft": "0.25rem",
            }
          }
          type="submit"
        >
           Clear
        </button>
      </div>
    </div>
  </form>
</div>
`;
